### PR TITLE
Make table manager retention options configurable in helm chart values

### LINKED ISSUE #8676: table_manager - Dynamo DB - Impossible to change provisioning capacity
**Describe the bug**
Hello,
i am trying to deploy the loki-grafana chart using S3 and DynamoDB. I see that the "Read Capacity Mode" is set to 300 and the "Write Capacity Mode" is set to 1000.

I see from the documentation that it is possible to configure the DynamoDB Provisioning:

https://grafana.com/docs/loki/latest/operations/storage/table-manager/

But in the values.yaml of the chart the section "table_manager" (inside config) is not customizable and it is fixed:

```
    table_manager:
      retention_deletes_enabled: false
      retention_period: 0
```

**To Reproduce**
1. Started Loki (SHA or version) --> Chart Version: 4.8.0, Application Version: 2.7.3

**Expected behavior**
It should be possible customize the table_manager section as described in the documentation above:
```

table_manager:
  index_tables_provisioning:
    provisioned_write_throughput: 10
    provisioned_read_throughput: 10
  chunk_tables_provisioning:
    provisioned_write_throughput: 10
    provisioned_read_throughput: 10
```

**Environment:**
 - Infrastructure: [e.g., Kubernetes, bare-metal, laptop]: Kubernetes EKS
 - Deployment tool: [e.g., helm, jsonnet]: helm

### PR DESCRIPTION
**Pull request by NilsGriebner**
**What this PR does / why we need it**:

Configuration options for table manager retention are hard-coded in the helm chart values at the moment so that it's not possible to enable retention deletes.

**Which issue(s) this PR fixes**:
Fixes #8676

**Special notes for your reviewer**:
-

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: production/helm/loki/values.yaml [hunk 1]
```diff
@@ -813,6 +813,13 @@ tableManager:
   nodeSelector: {}
   # -- Tolerations for table-manager pods
   tolerations: []
+  # -- Configuration for table-manager
+  config:
```
Comment (MichelHollands) [inline production/helm/loki/values.yaml:817]: This doesn't really have to be in a config block.

File: production/helm/loki/CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,10 @@ Entries should include a reference to the pull request that introduced the chang
 
 [//]: # (<AUTOMATED_UPDATES_LOCATOR> : do not remove this line. This locator is used by the CI pipeline to automatically create a changelog entry for each new Loki release. Add other chart versions and respective changelog entries bellow this line.)
 
+## 5.6.3
```
Comment (MichelHollands) [inline production/helm/loki/CHANGELOG.md:16]: Someone else already took 5.6.3. Please make it 5.6.4.

File: production/helm/loki/values.yaml (hunk 1 above)
Comment (NilsGriebner) [reply production/helm/loki/values.yaml:817]: I rebased and made the changes you requested but the PR is not updating?! Maybe something is wrong with Github? I'll investigate further.

Comment (JStickler) [reply production/helm/loki/values.yaml:817]: GitHub status page is looking colorful. I suspect the problem is on the GitHub side.

https://www.githubstatus.com/

Comment (NilsGriebner) [reply production/helm/loki/values.yaml:817]: Ahh makes sense. I was pulling my hair out on this already :smile: Thanks for the hint!

Comment (NilsGriebner) [reply production/helm/loki/values.yaml]: This is now fixed.

File: production/helm/loki/CHANGELOG.md (hunk 2 above)
Comment (NilsGriebner) [reply production/helm/loki/CHANGELOG.md]: Fixed.

**Review by JStickler — APPROVED** [review]
[docs team] LGTM

**Review by MichelHollands — APPROVED** [review]
LGTM

**Pull request merged**